engine/jobmaster/dm/dm_jobmaster.go

```diff
@@ -51,6 +51,9 @@ type JobMaster struct {
 	// only use when init
 	// it will be outdated if user update job cfg.
 	initJobCfg *config.JobCfg
+	// taskID -> FinishedTaskStatus
+	// worker exists after finished, so we need record the finished status for QueryJobStatus
+	finishedStatus sync.Map
```

Comment:
Worth writing in comment that `finishedStatus` will be lost when jobmaster failover.

And can we use Task.Stage to mark a task is finished? So when failover, only the summary status is lost. And maybe we can also store summary status into Task